nucypher/config/storages.py

```diff
@@ -539,19 +543,16 @@ def __init__(self, *args, **kwargs):
     #         shutil.rmtree(self.__temp_certificates_dir, ignore_errors=True)
 
     def initialize(self) -> bool:
+        # Root
+        self.__temp_root_dir = tempfile.mkdtemp(prefix="nucypher-tmp-nodes-")
+        self.root_dir = self.__temp_root_dir
+
         # Metadata
-        self.__temp_metadata_dir = tempfile.mkdtemp(prefix="nucypher-tmp-nodes-")
+        self.__temp_metadata_dir = str(Path(self.__temp_root_dir) / "metadata")
         self.metadata_dir = self.__temp_metadata_dir
 
         # Certificates
-        self.__temp_certificates_dir = tempfile.mkdtemp(prefix="nucypher-tmp-certs-")
+        self.__temp_certificates_dir = str(Path(self.__temp_root_dir) / "certs")
         self.certificates_dir = self.__temp_certificates_dir
 
-        return bool(os.path.isdir(self.metadata_dir) and os.path.isdir(self.certificates_dir))
-
-
-#
-# Node Storage Registry
-#
-NODE_STORAGES = {storage_class._name: storage_class
-                 for storage_class in NodeStorage.__subclasses__()}
+        return all(map(os.path.isdir, (self.root_dir, self.metadata_dir, self.certificates_dir)))
```

Comment:
The return value doesn't seem to be checked on use. May be better to raise an exception here, or just not check it at all.

Reply:
I agree!

Reply:
✅